designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,236 @@
+
+- **cert-manager**: Automates the management and issuance of TLS certificates, facilitating encrypted communication and, when configured with mTLS, adding a layer of authentication.
+  Currently, we leverage on cert-manager when webhooks are scaffold. So, the proposal idea would be to allow users enable the cert-manager for the metrics such as it is provided
+  and required for webhook feature. However, it MUST be optional. One of the goals of Kubebuilder is make it easier for new users, therefore new users should
+  not need to deal with cert-manager by default or have the need to install it to just quick start.
+
+- **Controller-Runtime Feature (PR #2407)[cr-pr]**:  introduces options for metrics that allow serving metrics via TLS and using authentication and authorization.
+It seems fit very well to our requirements.
+
+While this setup does not replicate [kube-rbac-proxy's][kube-rbac-proxy] method of leveraging Kubernetes RBAC for authentication
+and authorization directly, it achieves a comparable level of security through a combination of network isolation and encrypted communication.
+
+### Phase 3: When kube-rbac-proxy be accepted under the umbrella
+
+Once kube-rbac-proxy is included in the Kubernetes umbrella,
+Kubebuilder maintainers can support its integration through a [plugin](https://kubebuilder.io/plugins/plugins).
+This enables a seamless way to incorporate kube-rbac-proxy into Kubebuilder scaffolds,
+allowing users to run:
+
+```sh
+kubebuilder init|edit --plugins="kube-rbac-proxy/v1"
+```
+
+So that, the plugin could use the [plugin/util](../pkg/plugin/util) lib provide
+to comment (We can add a method like the [UncommentCode](https://github.com/kubernetes-sigs/kubebuilder/blob/72586d386cfbcaecea6321a703d1d7560c521885/pkg/plugin/util/util.go#L102))
+the patches in the `config/default/kustomization` and disable the default network policy used within
+and [replace the code](https://github.com/kubernetes-sigs/kubebuilder/blob/72586d386cfbcaecea6321a703d1d7560c521885/pkg/plugin/util/util.go#L231)
+in the `main.go` bellow with in order to not use the controller-runtime
+feature instead.
+
+```go
+    ctrlOptions := ctrl.Options{
+		MetricsFilterProvider: filters.WithAuthenticationAndAuthorization,
+		MetricsSecureServing:  true,
+	}
+```
+
+### Proof of Concept
+
+- **(Phase 1)NetworkPolicies:** https://github.com/kubernetes-sigs/kubebuilder/pull/3853
+- Example of Controller-Runtime new feature to protect Metrics Endpoint: https://github.com/sbueringer/controller-runtime-tests/tree/master/metrics-auth
+
+### Risks and Mitigations
+
+#### Loss of Previously Promoted Images
+
+The transition to the new shared infrastructure for Kubernetes SIG projects has rendered us unable to automatically
+build and promote images as before. The process only works for projects under the umbrella.
+However, the k8s-infra maintainers are assisting in manually transferring these images
+to the new [registry.k8s.io][registry.k8s.io].
```

Comment:
The images are manually added there now; https://explore.ggcr.dev/?repo=gcr.io%2Fk8s-staging-kubebuilder%2Fkube-rbac-proxy

Reply:
added so closing